File: solutions/1614/main.py
class Solution:
    def maxDepth(self, s: str) -> int:
        maxDepth = 0
        leftP, rightP = 0, 0
        for c in s: 
            if c == "(": 
                leftP += 1 
            elif c == ")":  
                diff = leftP - rightP 
                maxDepth = max(diff, maxDepth)
                rightP += 1
        return maxDepth

if __name__ == "__main__": 
    s = "(1+(2*3)+((8)/4))+1"
    sol = Solution()
    assert sol.maxDepth(s) == 3

File: solutions/1614/main.cpp
/* ************************************************************************
> File Name:     main.cpp
> Author:        Sean
> Project:    LeetCode
> Created Time:  Fri Jan  7 08:57:33 2022
> Description:   
 ************************************************************************/
#include <string> 
#include <gtest/gtest.h> 

using namespace std; 

class Solution {
public:
    int maxDepth(string s) {
        int maxDepth = 0;
        int left = 0, right = 0;
        for (auto& c : s) {
            if (c == '(') left++;
            if (c == ')') {
                maxDepth = max(left - right, maxDepth);
                right++;
            }
        }
        return maxDepth;
    }
};

TEST(Test1614, SimpleTest) {
    string s = "(1+(2*3)+((8)/4))+1";
    Solution sol; 
    EXPECT_EQ(sol.maxDepth(s), 3);
}

int main() {
    testing::InitGoogleTest();
    return RUN_ALL_TESTS();
}
